docker-official-images/internal/config/config_test.go

```diff
@@ -0,0 +1,344 @@
+			cfg: config.ActiveVersionConfig{
+				Branch: "main",
+				Type:   "", // Missing
+			},
+			wantErr: true,
+		},
+		{
+			name: "Missing both Branch and Type",
+			cfg: config.ActiveVersionConfig{
+				Branch: "", // Missing
+				Type:   "", // Missing
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.cfg.Validate() // This method doesn't exist yet!
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Validate() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			// Optional: If you want to check for specific error messages later,
+			// you can add checks like:
+			// if tt.wantErr && !strings.Contains(err.Error(), "branch is mandatory") { ... }
+		})
+	}
+}
+
+// TestValidationTypeField tests that the Type field has an allowed value.
+func TestValidationTypeField(t *testing.T) {
+	tests := []struct {
+		name string
+		cfg  config.ActiveVersionConfig
+		wantErr bool
+	}{
+		{
+			name: "Valid Type: communityBuild",
+			cfg: config.ActiveVersionConfig{
+				Branch: "main",
+				Type:   "communityBuild",
+			},
+			wantErr: false,
+		},
+		{
+			name: "Valid Type: commercialEditions",
+			cfg: config.ActiveVersionConfig{
+				Branch: "main",
+				Type:   "commercialEditions",
+			},
+			wantErr: false,
+		},
+		{
+			name: "Valid Type: legacy",
+			cfg: config.ActiveVersionConfig{
+				Branch: "main",
+				Type:   "legacy",
+			},
+			wantErr: false,
+		},
+		{
+			name: "Invalid Type: unknown",
+			cfg: config.ActiveVersionConfig{
+				Branch: "main",
+				Type:   "unknown", // Invalid
+			},
+			wantErr: true,
+		},
+		{
+			name: "Invalid Type: custom",
+			cfg: config.ActiveVersionConfig{
+				Branch: "main",
+				Type:   "custom", // Invalid
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.cfg.Validate() // This method still doesn't exist!
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Validate() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			// Optional: check for specific error messages later if needed
+		})
+	}
+}
+
+// TestParseConfigFile tests the parsing of a JSON configuration file.
+func TestParseConfigFile(t *testing.T) {
+	// 1. Create a temporary directory and a JSON file within it
+	tempDir, err := os.MkdirTemp("", "config-test-*")
+	if err != nil {
+		t.Fatalf("Failed to create temp dir: %v", err)
+	}
+	// Clean up the temporary directory when the test finishes
+	defer os.RemoveAll(tempDir)
+
+	configFilePath := filepath.Join(tempDir, "test_config.json")
+	jsonContent := `
+	[
+		{
+			"branch": "feature/new",
+			"type": "communityBuild"
+		},
+		{
+			"branch": "pathlta",
+			"type": "commercialEditions",
+			"commitSha": "deadbeef"
+		}
+	]`
+
+	// Write the JSON content to the temporary file
+	err = os.WriteFile(configFilePath, []byte(jsonContent), 0644) // 0644 is standard file permissions
+	if err != nil {
+		t.Fatalf("Failed to write temp config file: %v", err)
+	}
```

Comment:
If you really want to dynamically create config tests, this can be a helper function: given a string, it returns the path of the temp file created.

I would avoid doing that, and I would split the `config.ParseConfigFile` instead, because I (speak I) am not interested in the create-file-open-file-read-file logic: I assume that it works. I am interested only if given `{"this":"json"}`, `config.ParseConfigFile` will fail because it isn't a valid config file.